File: Week 01/id_113/LeetCode641.py
class MyCircularDeque:

    def __init__(self, k: int):
        """
        Initialize your data structure here. Set the size of the deque to be k.
        """
        self.size = k
        self.array = []

    def insertFront(self, value: int) -> bool:
        """
        Adds an item at the front of Deque. Return true if the operation is successful.
        """
        if not self.isFull():
            self.array.insert(0, value)
            return True

        return False

    def insertLast(self, value: int) -> bool:
        """
        Adds an item at the rear of Deque. Return true if the operation is successful.
        """
        if not self.isFull():
            self.array.append(value)
            return True
        return False

    def deleteFront(self) -> bool:
        """
        Deletes an item from the front of Deque. Return true if the operation is successful.
        """
        if not self.isEmpty():
            self.array.pop(0)
            return True
        return False

    def deleteLast(self) -> bool:
        """
        Deletes an item from the rear of Deque. Return true if the operation is successful.
        """
        if not self.isEmpty():
            self.array.pop()
            return True
        return False

    def getFront(self) -> int:
        """
        Get the front item from the deque.
        """
        if not self.isEmpty():
            return self.array[0]
        return -1

    def getRear(self) -> int:
        """
        Get the last item from the deque.
        """
        if not self.isEmpty():
            count = len(self.array)
            return self.array[count - 1]
        return -1

    def isEmpty(self) -> bool:
        """
        Checks whether the circular deque is empty or not.
        """
        if len(self.array) == 0:
            return True
        return False

    def isFull(self) -> bool:
        """
        Checks whether the circular deque is full or not.
        """
        if len(self.array) == self.size:
            return True
        return False


# Your MyCircularDeque object will be instantiated and called as such:
if __name__ == "__main__":
    obj = MyCircularDeque(k)
    param_1 = obj.insertFront(value)
    param_2 = obj.insertLast(value)
    param_3 = obj.deleteFront()
    param_4 = obj.deleteLast()
    param_5 = obj.getFront()
    param_6 = obj.getRear()
    param_7 = obj.isEmpty()
    param_8 = obj.isFull()

File: Week 01/id_343/LeetCode_641.go

type MyCircularDeque struct {
    Size int
    Values []int
}


/** Initialize your data structure here. Set the size of the deque to be k. */
func Constructor(k int) MyCircularDeque {
    return MyCircularDeque{Size:k}
}


/** Adds an item at the front of Deque. Return true if the operation is successful. */
func (this *MyCircularDeque) InsertFront(value int) bool {
    if (len(this.Values) == this.Size) {
        return false
    } 
    this.Values = append([]int{value}, this.Values...)
    return true
}


/** Adds an item at the rear of Deque. Return true if the operation is successful. */
func (this *MyCircularDeque) InsertLast(value int) bool {
    if (len(this.Values) == this.Size) {
        return false
    } 
    this.Values = append(this.Values, value)
    return true
}


/** Deletes an item from the front of Deque. Return true if the operation is successful. */
func (this *MyCircularDeque) DeleteFront() bool {
    if (len(this.Values) == 0) {
        return false
    } 
    this.Values = this.Values[1:]
    return true
}


/** Deletes an item from the rear of Deque. Return true if the operation is successful. */
func (this *MyCircularDeque) DeleteLast() bool {
    if (len(this.Values) == 0) {
        return false
    } 
    this.Values = this.Values[:len(this.Values)-1]
    return true
}

/** Get the front item from the deque. */
func (this *MyCircularDeque) GetFront() int {
    if len(this.Values) == 0 {
        return -1
    }
    return this.Values[0]
}


/** Get the last item from the deque. */
func (this *MyCircularDeque) GetRear() int {
    if len(this.Values) == 0 {
        return -1
    }
    return this.Values[len(this.Values)-1]
}


/** Checks whether the circular deque is empty or not. */
func (this *MyCircularDeque) IsEmpty() bool {
    return len(this.Values) == 0
}


/** Checks whether the circular deque is full or not. */
func (this *MyCircularDeque) IsFull() bool {
    return len(this.Values) == this.Size
}
